compiler/damlc/tests/src/DA/Test/DataDependencies.hs

```diff
@@ -286,6 +286,70 @@ tests tools@Tools{damlc,validate,oldProjDar} = testGroup "Data Dependencies" $
           validate $ projc </> "projc.dar"
     | (depLfVer, targetLfVer) <- lfVersionTestPairs
     ] <>
+    [ testCaseSteps ("Cross Daml-LF version with double data-dependency from old SDK: " <> LF.renderVersion depLfVer <> " -> " <> LF.renderVersion targetLfVer) $
+        -- Given a dar "Old" built with an older SDK, this tests that a project
+        -- which depends on "Old" through different paths on its dependency graph
+        -- will not end up with multiple copies of daml-prim and daml-stdlib
+        -- from the older SDK, which would prevent it from compiling.
+        \step -> withTempDir $ \tmpDir -> do
+            let proja = tmpDir </> "proja"
+            let projb = tmpDir </> "projb"
+
+            step "Build proja"
+            createDirectoryIfMissing True (proja </> "src")
+            writeFileUTF8 (proja </> "src" </> "A.daml") $ unlines
+                [ "module A where"
+                , "import Old ()"
+                , "template T"
+                , "  with"
+                , "    party : Party"
+                , "  where"
+                , "    signatory party"
+                ]
+            writeFileUTF8 (proja </> "daml.yaml") $ unlines
+                [ "sdk-version: " <> sdkVersion
+                , "name: proja"
+                , "version: 0.0.1"
+                , "source: src"
+                , "dependencies: [daml-prim, daml-stdlib]"
+                , "data-dependencies:"
+                , " - " <> show oldProjDar
```

Comment:
Did you verify that the issue (without your fix) happens when you use this specific dar (and these sources)? 

If it doesn't, I think it would be good to add a 1.18.1 dar for this purpose (and it's also good in general to ensure continuity from daml 1.x to 2.x).

Reply:
Yes I ran the test with and without the fix a few times, I can confirm it does fail without the fix